File: src/main/java/se/lars/types/Listing.java
package se.lars.types;


import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import org.immutables.value.Value;

import java.util.*;


@Value.Immutable
@JsonSerialize(as = ImmutableListing.class)
@JsonDeserialize(as = ImmutableListing.class)
public interface Listing
{
    String id();
    Optional<String> name();
    Optional<String> type();
    Optional<String> currencyCode();
    Optional<String> longName();
    Optional<String> isinCode();
    Optional<String> micCode();
    Optional<Date> listingDate();
    OptionalInt roundLot();
    Optional<Reference> getIssuer();
}

File: src/main/kotlin/se/lars/types/Listing.kt
package se.lars.types


import java.util.*


data class Listing (
    val id: String,
    val name: String?,
    val type: String?,
    val currencyCode: String?,
    val longName: String?,
    val isinCode: String?,
    val micCode: String?,
    val listingDate: Date?,
    val roundLot: Int?,
    val issuer: Reference?
)
